### PR TITLE
Refactor lifi class

### PR DESCRIPTION
**Pull request by paszkowskiDamian**

### CODE REVIEW

File: packages/app/src/domain/exchanges/lifi/lifi.ts [hunk 1]
```diff
@@ -50,17 +47,40 @@ interface ReverseQuoteRequestParams {
 }
 
 export class LiFi {
-  private baseUrl = 'https://li.quest'
-  private chainId: number
-  private userAddress: CheckedAddress
+  private readonly baseUrl = 'https://li.quest'
+  private readonly pathName = 'v1/quote/contractCalls'
```
Comment (yivlad) [inline packages/app/src/domain/exchanges/lifi/lifi.ts:51]: Good idea extracting those, can you extract he one for `buildQuoteUrl` as well?

File: packages/app/src/domain/exchanges/lifi/lifi.ts [hunk 2]
```diff
@@ -50,17 +47,40 @@ interface ReverseQuoteRequestParams {
...
+    const response = await fetch(url)
+    if (!response.ok) {
+      throw new Error('Error fetching exchange quote')
+    }
+    const result = await response.json()
+
+    return this.parseQuoteResponse(result)
+  }
+
+  getKey(chainId: number, userAddress: CheckedAddress): string {
```
Comment (yivlad) [inline packages/app/src/domain/exchanges/lifi/lifi.ts:78]: I think we should get rid of this method

Comment (paszkowskiDamian) [reply packages/app/src/domain/exchanges/lifi/lifi.ts:78]:
```
 return queryOptions<SwapRequest>({
    queryKey: ['liFiTxData', client.getKey(chainId, userAddress), type, fromToken, toToken, amount, maxSlippage],
    queryFn: async (): Promise<SwapRequest> => {
```

It used in `query.ts` do you think I should just inline this in there or create separate function for that?

File: packages/app/src/domain/exchanges/lifi/lifi.ts [hunk 3]
```diff
@@ -50,17 +47,40 @@ interface ReverseQuoteRequestParams {
 }
 
 export class LiFi {
```
Comment (oskarvu) [inline packages/app/src/domain/exchanges/lifi/lifi.ts:49]: Should we change it to `LiFiClient`?

**Review by yivlad — APPROVED** [review]

**Review by oskarvu — APPROVED** [review]

**Pull request merged**